grails-app/views/report/showInventoryByLocationReport.gsp

```diff
@@ -35,8 +35,27 @@
                                                   data-placeholder=" "
                                                   activityCode="${org.pih.warehouse.core.ActivityCode.MANAGE_INVENTORY}"/>
                             </div>
+                            <label><warehouse:message code="category.label"/></label>
+                            <p>
+                                <g:selectCategory
+                                        id="category" multiple="true"
+                                        class="chzn-select-deselect filter"
+                                        data-placeholder="${g.message(code: 'category.selectCategory.label', default: 'Select a category')}"
+                                        name="category"
+                                        noSelection="['':'']"
+                                        value="${params?.category instanceof String ? params?.category : params?.category?.collect { it }}"
```

Comment:
could you elaborate on this? 
It seems like the fallback's collect does not have any effect:
```groovy
params?.category?.collect { it }
```
Isn't it equivalent to JS'
```javascript
array.map(x => x)
```
?

Reply:
It doesn't work without .collect { it }. Now, when I know about `params.list(...)` I changed this part of code to this solution and now It is simplified. It is hard to say, why I doesn't work. I saw { it.id } above and I decided to try something similar and it started working after this weird change.